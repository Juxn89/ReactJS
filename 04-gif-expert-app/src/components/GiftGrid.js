import React from 'react';
import { Config } from '../commun';

const GiftGrid = ({category}) => {
    const getGift = async () => {
        const url = `https://api.giphy.com/v1/gifs/search?q=Samurai X&limit=10&api_key=${Config.apiKeyGiphy}`;
        const resp = await fetch(url);
        const {data} = await resp.json();
        const gifts = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });
        console.log(gifts);
    }

    getGift();

    return (
        <>
            <h3>{category}</h3>
        </>
    );
}

export default GiftGrid;