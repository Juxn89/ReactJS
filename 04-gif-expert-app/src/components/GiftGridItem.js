import React from "react";

export const GiftGridItem = ({id, title, url}) => {
    console.log(id, title, url);
    return (
        <div className="animate__animated animate__bounce animate__fadeIn">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    );
}