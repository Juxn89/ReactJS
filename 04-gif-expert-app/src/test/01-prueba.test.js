import React from 'react';
import { GiftGridItem } from '../components/GiftGridItem'
import { shallow } from "enzyme";

describe('Test on <GiftGridItem />', () => {
    const title = 'Un t√≠tulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GiftGridItem title={title} url = {url}/> );

    test('Sould show the component', () =>{
        expect(wrapper).toMatchSnapshot();
    });

    test('Should have a paragraph with the title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('should have image equals to url and alt from props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Should have animate_fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    });
});