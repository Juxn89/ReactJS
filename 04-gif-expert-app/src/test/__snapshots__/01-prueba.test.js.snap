// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GiftGridItem /> Sould show the component 1`] = `
<div
  className="animate__animated animate__bounce animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
